applications/GeoMechanicsApplication/custom_elements/transient_Pw_line_element.h

```diff
@@ -0,0 +1,419 @@
+    {
+        for (const auto& node : GetGeometry()) {
+            KRATOS_ERROR_IF_NOT(node.SolutionStepsDataHas(rVariable))
+                << "Missing variable " << rVariable.Name() << " on node " << node.Id() << std::endl;
+        }
+    }
+
+    void CheckHasDofsFor(const Variable<double>& rVariable) const
+    {
+        for (const auto& node : GetGeometry()) {
+            KRATOS_ERROR_IF_NOT(node.HasDofFor(rVariable))
+                << "Missing degree of freedom for " << rVariable.Name() << " on node " << node.Id()
+                << std::endl;
+        }
+    }
+
+    void CheckProperties() const
+    {
+        CheckProperty(DENSITY_WATER);
+        CheckProperty(DENSITY_SOLID);
+        CheckProperty(POROSITY);
+        CheckProperty(BULK_MODULUS_SOLID);
+        CheckProperty(BULK_MODULUS_FLUID);
+        CheckProperty(DYNAMIC_VISCOSITY);
+        CheckProperty(BIOT_COEFFICIENT);
+        CheckProperty(PERMEABILITY_XX);
+    }
+
+    void CheckProperty(const Kratos::Variable<double>& rVariable) const
+    {
+        KRATOS_ERROR_IF_NOT(GetProperties().Has(rVariable))
+            << rVariable.Name() << " does not exist in the thermal element's properties" << std::endl;
+        KRATOS_ERROR_IF(GetProperties()[rVariable] < 0.0)
+            << rVariable.Name() << " has an invalid value at element " << Id() << std::endl;
+    }
+
+    void CheckProperty(const Kratos::Variable<std::string>& rVariable, const std::string& rName) const
+    {
+        KRATOS_ERROR_IF_NOT(GetProperties().Has(rVariable))
+            << rVariable.Name() << " does not exist in the thermal element's properties" << std::endl;
+        KRATOS_ERROR_IF_NOT(GetProperties()[rVariable] == rName)
+            << rVariable.Name() << " has a value of (" << GetProperties()[rVariable]
+            << ") instead of (" << rName << ") at element " << Id() << std::endl;
+    }
+
+    void CheckForNonZeroZCoordinateIn2D() const
+    {
+        if constexpr (TDim == 2) {
+            const auto& r_geometry = GetGeometry();
+            auto        pos        = std::find_if(r_geometry.begin(), r_geometry.end(),
+                                                  [](const auto& node) { return node.Z() != 0.0; });
+            KRATOS_ERROR_IF_NOT(pos == r_geometry.end())
+                << " Node with non-zero Z coordinate found. Id: " << pos->Id() << std::endl;
+        }
+    }
+
+    static void AddContributionsToLhsMatrix(MatrixType& rLeftHandSideMatrix,
+                                            const BoundedMatrix<double, TNumNodes, TNumNodes>& rPermeabilityMatrix,
+                                            const BoundedMatrix<double, TNumNodes, TNumNodes>& rCompressibilityMatrix,
+                                            double DtPressureCoefficient)
+    {
+        rLeftHandSideMatrix = rPermeabilityMatrix;
+        rLeftHandSideMatrix += (DtPressureCoefficient * rCompressibilityMatrix);
```

Comment:
I think this could be a one-liner:
```suggestion
        rLeftHandSideMatrix = rPermeabilityMatrix + DtPressureCoefficient * rCompressibilityMatrix;
```